js/compiler/compiler.go

```diff
@@ -86,10 +87,13 @@ var (
 	globalBabel        *babel        // nolint:gochecknoglobals
 )
 
+const sourceMapURLFromBabel = "k6://internal-should-not-leak/file.map"
```

Comment:
This is literally a constant that babel sets in the file which tells goja (in this case) where the sourcemap is. Goja asks k6 to get this and we return the sourcemap babel generated. It's just a particular *very* unlikely value for a real sourcemap URL ;)

Reply:
Partially out of curiosity: what is the worst that can happen if someone does use this internal URL deliberately for external sourcemap?

Reply:
The code returning the sourcemap to goja matches first on it and returns an internal field that get's set when babel transforms. If babel hasn't ran the field is nil - so nothing, if it has it would've set this url.